tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -41,13 +41,19 @@ spec:
           # oras has very limited support for selecting the right auth entry,
           # so create a custom auth file with just one entry.
           DEST_AUTH_FILE=$(mktemp)
-          select-oci-auth "$4:$5" > "$DEST_AUTH_FILE"
+          registry=$(echo "$4" | cut -d '/' -f 1)
+          select-oci-auth "$4" | jq -c \
+            '.auths."'$4'" = .auths."'$registry'" | del(.auths."'$registry'")' > "$DEST_AUTH_FILE"
 
           # We need this default value for $7 when oras_args is equal to ()
           destination_digest=$(oras resolve --registry-config "$DEST_AUTH_FILE" ${7:-} "$4:$5" || true)
 
           if [[ "$destination_digest" != "$1" || -z "$destination_digest" ]]; then
             printf '* Pushing component: %s to %s:%s\n' "$2" "$4" "$5"
+            # Create a combined auth file to enable partial oci matches to work
+            COMBINED_AUTH_FILE=$(mktemp)
+            jq -s 'reduce .[] as $item ({}; . * $item)' $SOURCE_AUTH_FILE $DEST_AUTH_FILE > $COMBINED_AUTH_FILE
+            export DOCKER_CONFIG="$COMBINED_AUTH_FILE"
```

Comment:
From my brief experiment with this, it seems that DOCKER_CONFIG should point to a directory (normally ~/.docker) that will contain config.json. So you can probably do something like:
```sh
export DOCKER_CONFIG=(mktemp -d)
jq ... > $DOCKER_CONFIG/config.json
```

Reply:
ah, good to know, thanks